src/platform.ts

```diff
@@ -79,10 +92,46 @@ export class MideaPlatform implements DynamicPlatformPlugin {
     // If we have configuration indexed by ID use that, if not use IP address.
     const deviceConfig = this.platformConfig.devices.find((device: DeviceConfig) => device.id === device_info.id);
     if (deviceConfig) {
+      // keep track of discovered devices
+      this.discoveredDevices[device_info.id] = true;
+      // Override name with that provided in the config.json settings
       device_info.name = deviceConfig.name ?? device_info.name;
       this.addDevice(device_info, deviceConfig);
     } else {
-      this.log.info(`[${device_info.name} | ${device_info.ip}}] Device is not added to config.`);
+      this.log.warn(`[${device_info.name} | ${device_info.ip}}] New device discovered, you must add it to the config.json settings.`);
+    }
+  }
+
+  /*********************************************************************
+   * discoveryComplete
+   * Function called by the 'complete' on handler.
+   */
+  private discoveryComplete() {
+    this.log.debug(`Discovery complete, check for missing devices`);
+    // Check if network broadcasting found all devices that user configured.  If not then
+    // we have to handle those.
+    let missingDevices = 0;
+    Object.values(this.platformConfig.devices).forEach((device) => {
+      if (!this.discoveredDevices[device.id]) {
+        // This device was not found by network discovery.
+        missingDevices++;
+        if (this.discoveredDevices[device.id] === undefined) {
+          this.discoveredDevices[device.id] = false;
+          this.log.warn(`[${device.name}] Device not found (id: ${device.id}), will retry every ${this.discoveryInterval} seconds`);
+        } else {
+          this.log.debug(`[${device.name}] Device not found (id: ${device.id}), will retry in ${this.discoveryInterval} seconds`);
+        }
+      }
+    });
+    if (missingDevices > 0) {
+      // Some devices not found. Keep retrying periodically until all are found.
+      setTimeout(() => {
+        missingDevices = 0;
+        // on repeat attempts only retry once (so two broadcasts sent), 3000 miliseconds apart.
+        this.discover.startDiscover(1, 3000);
```

Comment:
Maybe I would set the timeout to even bigger, like every 5 minutes, although I don't know how heavy the operation to send discovery messages on all interfaces.

Reply:
I wondered about that.  But I think that a broadcast once a minute is pretty lightweight on a LAN.  I pondered making it configurable, but you have to draw the line somewhere in what makes sense to allow a user to tinker with.  Happy to change it to a longer duration if you think it better.